src/dataflow-types/src/types.rs

```diff
@@ -1369,42 +1369,111 @@ pub mod sources {
     /// AWS SDK so that we can implement `Serialize` and `Deserialize`.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
     pub struct AwsConfig {
-        /// The optional static credentials to use.
-        ///
-        /// If unset, credentials should instead be obtained from the environment.
-        pub credentials: Option<AwsCredentials>,
+        /// AWS Credentials, or where to find them
+        pub credentials: AwsCredentials,
         /// The AWS region to use.
-        pub region: String,
+        ///
+        /// Uses the default region (looking at env vars, config files, etc) if not provideed.
+        pub region: Option<String>,
+        /// The AWS role to assume
+        pub role: Option<AssumeRole>,
         /// The custom AWS endpoint to use, if any.
         pub endpoint: Option<SerdeUri>,
     }
 
-    /// Static AWS credentials for a source or sink.
+    /// AWS credentials for a source or sink.
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub enum AwsCredentials {
+        /// Look for credentials using the [default credentials chain][credchain]
+        ///
+        /// [credchain]: aws_config::default_provider::credentials::DefaultCredentialsChain
+        Default,
+        /// Load credentials using the given named profile
+        Profile { profile_name: String },
+        /// Use the enclosed static credentials
+        Static {
+            access_key_id: String,
+            secret_access_key: String,
+            session_token: Option<String>,
+        },
+    }
+
+    /// A role for Materialize to assume when performing AWS API calls.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct AwsCredentials {
-        pub access_key_id: String,
-        pub secret_access_key: String,
-        pub session_token: Option<String>,
+    pub struct AssumeRole {
+        /// The Amazon Resource Name of the Role to assume.
+        pub arn: String,
+        /// The External ID for this customer Materialize provides during role assumption.
+        ///
+        /// <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html>
+        pub external_id: Option<String>,
```

Comment:
I'm loathe to expand the scope of this PR further, but ideally we'd change this to be a parameter on `load`, perhaps via a type like 

```
enum AwsExternalId {
    UserSpecified(String),
    ISwearTheUserDidNotProvideOne,
}
```

to make it _really_ clear that you 100% must plumb in whatever value the user specified. Mind filing a follow-up issue for that?

Reply:
Actually, done! #10522